pkg/cmd/generate/main.go

```diff
@@ -0,0 +1,164 @@
+package generate
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	v1 "github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	"github.com/jaegertracing/jaeger-operator/pkg/strategy"
+	"github.com/jaegertracing/jaeger-operator/pkg/version"
+	k8s_json "k8s.io/apimachinery/pkg/runtime/serializer/json"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+// NewGenerateCommand starts the Jaeger Operator
+func NewGenerateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "generate",
+		Short: "Generate YAML manifests from Jaeger CRD",
+		Long: `Generate YAML manifests from Jaeger CRD.
+
+Defaults to reading Jaeger CRD from standard input and writing it to standard output, override with --cr <filename> and --output <filename>.`,
+		RunE: generate,
+	}
+
+	cmd.Flags().String("cr", "/dev/stdin", "Input Jaeger CRD")
+	viper.BindPFlag("cr", cmd.Flags().Lookup("cr"))
+
+	cmd.Flags().String("output", "/dev/stdout", "Where to print the generated YAML documents")
```

Comment:
note that the operator logs its operations. Stdout will be mixed with logs and the result.

Reply:
Actually they log to stderr, so I think this is OK